.github/workflows/tekton_task_tests.yaml

```diff
@@ -0,0 +1,55 @@
+---
+name: Tekton Task Tests
+on:
+  pull_request:
+    branches: ['main']
+    paths:
+      - 'catalog/task/**'
+jobs:
+  run-tekton-task-tests:
+    name: Run Tekton Task tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.4.0
+      - name: Check cluster info
+        run: |
+          kubectl cluster-info
+      - name: Install Tekton
```

Comment:
how long does this task run? shouldn't it have an overall timeout?

Reply:
`kubectl apply` (1) is instant. Then waiting for the pods to appear (2) will typically take under 1 s. And finally it will typically take around 20 s for the pods to be ready (3).

So I don't think there is a significant risk that the while loop (2) will take forever. The biggest risk is that the pods will not be ready within 120 s and then the task testing will blow up in the later steps.

Maybe I should add another check of the pod readiness at the end of this step and just fail if they're still not ready? WDYT?

Reply:
sure, that sounds better. I am just a little hesitant to have while sleep loops with no timeout. Outages or whatever else may lead to things hanging so a timeout would be nice

Reply:
Actually, the `tekton wait` will return a non-zero exit code if the timeout is exceeded, so we're fine there. Doing a timeout for a `while` loop in bash is [a little cumbersome](https://stackoverflow.com/questions/12321469/retry-a-bash-command-with-timeout) so instead I set up a step timeout of 10 minutes (default is 360). Please check.